### PR TITLE
Summary Widget

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <gordon.smith@lexisnexis.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage @mlzummo @dtsnell4 Please Review.
Note:  Check out the new HTMLWidget GetBBox method.

File: src/common/Widget.js [hunk 1]
```diff
@@ -654,7 +651,9 @@
     };
 
     Widget.prototype.enter = function (domeNode, element, d) { };
+    Widget.prototype.preUpdate = function (domeNode, element, d) { };
```
Comment (mlzummo) [inline src/common/Widget.js]: i believe d is always undefined?

Comment (GordonSmith) [inline src/common/Widget.js]: Dunno - but its not changed in this PR

**Comment by mlzummo** [comment]
looks good

**Comment by GordonSmith** [comment]
Rebased

**Pull request merged**